### PR TITLE
Test cases: forms.py, fixes and test cases. Small refactoring of tests base

### PR DESCRIPTION
**Pull request by cyberstar**
Forms and utils test case for ``aldryn-reversions`` changed to inherit from ``HelperModelsObjectsSetupMixin`` which contains helper models objects setup.

``RecoverObjectWithTranslationForm`` is now covered with test cases (including edge cases):
* ``__init__``
* ``clean`` (validation)
* ``save`` (revert)

### CODE REVIEW

File: aldryn_reversion/tests/base.py [hunk 1]
```diff
@@ -102,3 +107,57 @@ def revert_to(self, object_with_revision, revision_number):
...
+        super(HelperModelsObjectsSetupMixin, self).setUp()
+        # prepare objects that can be used with utility methods
+        # simple model
+        self.simple_registered = self.create_with_revision(
+            SimpleRegistered,
+            **{'position': 1})
+
+        self.simple_no_admin = self.create_with_revision(
+            SimpleNoAdmin,
+            **{'position': 42})
```
Comment (intellectronica) [inline aldryn_reversion/tests/base.py]: Why are you writing it like this? The only reason to use `**` is if you have a dict variable. Since you know in advance the values, you could just write it thus:
```
self.simple_no_admin = self.create_with_revision(
    SimpleNoAdmin, position=42)
```

File: aldryn_reversion/tests/base.py [hunk 2]
```diff
@@ -102,3 +107,57 @@ def revert_to(self, object_with_revision, revision_number):
...
+
+        # translations
+        # note that WithTranslations will have 2 revisions, initial one
+        # and with 'de' translation
+        en_description = self.raw_text_string['en'].format(0)
+        de_description = self.raw_text_string['de'].format(0)
+        self.with_translation = self.create_with_revision(
+            WithTranslations,
+            **{'description': en_description})
```
Comment (intellectronica) [inline aldryn_reversion/tests/base.py]: See above

**Comment by intellectronica** [comment]
:+1: but please correct all the places where you include kwargs using `**` with a literal dict.

File: aldryn_reversion/tests/test_forms.py [hunk 3]
```diff
@@ -0,0 +1,214 @@
...
+            'revision': version.revision,
+            'obj': obj,
+            'version': version,
+            'resolve_conflicts': [],
+            'placeholders': [],
+        }
+
+    def build_unbound_form(self, arguments=None, **kwargs):
+        return self.form_class(
+            **(arguments if arguments is not None else kwargs))
```
Comment (mkoistinen) [inline aldryn_reversion/tests/test_forms.py]: Will this work for both args and kwargs? Genuine question.

Comment (cyberstar) [inline aldryn_reversion/tests/test_forms.py]: Well it expects arguments to be a dict, and if it is a dict then why not?
At least when i've checked that quickly it was looking like working.

Comment (mkoistinen) [inline aldryn_reversion/tests/test_forms.py]: I think it expects args to be a list/tuple and kwargs to be a dict
